fork/mklishevych/actions.js

```diff
@@ -0,0 +1,39 @@
+export const successAction = (text) => {
+    return {
+        type: 'ADD_NOTIFICATION_MIDDLEWARE',
+        callNotification: true,
+        text: text,
+        id: new Date().getTime(),
+        level: 'success'
+    };
+};
+
+export const errorAction = (text) => {
+    return {
+        type: 'ADD_NOTIFICATION_MIDDLEWARE',
+        callNotification: true,
+        text: text,
+        id: new Date().getTime(),
+        level: 'error'
+    };
+};
+
+export const errorByTypeAction = (type, error) => {
+    switch (type) {
+        case 'SIGNIN_USER':
```

Comment:
why not aggregate errors into one return?

Reply:
we can, but we still have to map error type that we recieve from the server to the error type that is used in the app. so we can change this to something like this:
```javascript
const map = { SIGNIN_USER: 'ERRORS_SIGNIN_USER', ... }
return map[type] ? { type: map[type], message: error } : undefined;
```

Reply:
well as I understand that low level shouldn't know nothing about action type and error handling

Reply:
again, not sure how the app uses dispatched payload, so I just did it backward compatible

Reply:
Ah, yes, I think you're right. This is low level code and it should just dispatch the generalized payload:
```javascript
export const errorByTypeAction = (type, error) => {
    return {
        type: 'RESPONSE_ERROR',
        errorType: type,
        error: error
    }
};
```
And then the old logic should be taken care of in the reducers based on errorType field